packages/core/src/codewhispererChat/tools/fsRead.ts

```diff
@@ -0,0 +1,138 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger/logger'
+import { readDirectoryRecursively } from '../../shared/utilities/workspaceUtils'
+import fs from '../../shared/fs/fs'
+
+const maxToolResponseSize = 30720 // 30KB
+
+enum OutputKind {
+    Text = 'text',
+    Json = 'json',
+}
+
+export interface InvokeOutput {
+    output: {
+        kind: OutputKind
+        content: string
+    }
+}
```

Comment:
I think we can pull these out to a common file since all tools will use the same interfaces

Reply:
Fixed.